include/deal.II/base/array_view.h

```diff
@@ -414,7 +414,8 @@ inline void
 ArrayView<ElementType, MemorySpaceType>::reinit(value_type *starting_element,
                                                 const std::size_t n_elements)
 {
-  *this = ArrayView(starting_element, n_elements);
+  this->starting_element = starting_element;
+  this->n_elements       = n_elements;
```

Comment:
How is this related to this PR?

Reply:
This is to work around a compile warning, as `clang-13` complains that this implicitly calls the copy constructor with this argument,
```
/home/kronbichler/deal/deal.II/include/deal.II/base/array_view.h:135:3: warning: definition of implicit copy assignment operator for 'ArrayView<dealii::VectorizedArray<double, 4>>' is deprecated because it has a user-provided copy constructor [-Wdeprecated-copy-with-user-provided-copy]
  ArrayView(const ArrayView<typename std::remove_cv<value_type>::type,
  ^
/home/kronbichler/deal/deal.II/include/deal.II/base/array_view.h:417:9: note: in implicit copy assignment operator for 'dealii::ArrayView<dealii::VectorizedArray<double, 4>>' first required here
  *this = ArrayView(starting_element, n_elements);
```
Unfortunately, I could not find a reduced test case to see whether the compile warning is correct or whether the compiler has a bug here, but I think it helps to avoid the warning here (and I do not see another way). In particular, I personally find the code actually becomes clearer without the indirection as one can more easily understand what happens. (Well, it needs one more line, but the additional copy just creates more work for the compiler.)